app/controllers/assignments_controller.rb

```diff
@@ -33,12 +36,25 @@ def index
   end
 
   def new
-    @assignment = Assignment.new
-    @assignment.course_id = @course.id
-    @assignment.due_date = (Time.now + 1.week).end_of_day.strftime("%Y/%m/%d %H:%M")
+    @files = Assignment.new
```

Comment:
They can have different properties, rather than specifying different behavior in the template or JS.